pkg/controller/networkpolicy/validate.go

```diff
@@ -615,6 +628,89 @@ func (v *antreaPolicyValidator) validateTierForPassAction(tier string, ingress,
 	return "", true
 }
 
+func (v *antreaPolicyValidator) validateEgressMulticastAddress(egressRule []crdv1alpha1.Rule) (string, bool) {
+	for _, r := range egressRule {
+		multicast := false
+		unicast := false
+		otherSelectors := false
+		for _, to := range r.To {
+			if to.IPBlock == nil {
+				continue
+			}
+			toIPAddr, _, err := net.ParseCIDR(to.IPBlock.CIDR)
+			if err != nil {
+				return fmt.Sprintf("invalid multicast groupAddress address (to.IPBlock.CIDR): %v", err.Error()), false
+			}
+			if toIPAddr.IsMulticast() {
+				multicast = true
+			} else {
+				unicast = true
+			}
+			if to.PodSelector != nil || to.NamespaceSelector != nil || to.Namespaces != nil ||
+				to.ExternalEntitySelector != nil || to.ServiceAccount != nil || to.NodeSelector != nil {
+				otherSelectors = true
+			}
+			if multicast && (*r.Action == crdv1alpha1.RuleActionPass || *r.Action == crdv1alpha1.RuleActionReject) {
+				return fmt.Sprintf("multicast does not support action Pass or Reject"), false
+			}
+		}
+		if multicast && unicast {
+			return fmt.Sprintf("can not set multicast groupAddress and unicast ip address at the same time"), false
+		}
+		if multicast && otherSelectors {
+			return fmt.Sprintf("can not set multicast groupAddress and selectors at the same time"), false
+		}
+	}
+	return "", true
+}
+
+func (v *antreaPolicyValidator) validateIngressMulticastAddress(ingressRule []crdv1alpha1.Rule) (string, bool) {
+	for _, r := range ingressRule {
+		for _, from := range r.From {
+			if from.IPBlock == nil {
+				continue
+			}
+			fromIPAddr, _, err := net.ParseCIDR(from.IPBlock.CIDR)
+			if err != nil {
+				return fmt.Sprintf("invalid IP address (to.IPBlock.CIDR): %v", err.Error()), false
+			}
+			if fromIPAddr.IsMulticast() {
+				return fmt.Sprintf("multicast is not support in ingress rule"), false
```

Comment:
but for ingress multicast traffic, group address should be `To.IPBlock.CIDR`, instead of `From.IPBlock.CIDR`, right? why we need to check `From` here?

Reply:
Do you mean from is the real scope, then to is multicast address? If so, I think we do not need to validate here, since `To` is not exposed in crd.

Reply:
removed the validation